javaparser-core/src/main/java/com/github/javaparser/printer/configuration/ConfigurablePrinter.java

```diff
@@ -0,0 +1,45 @@
+package com.github.javaparser.printer.configuration;
+
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.javaparser.printer.configuration.PrinterConfiguration.ConfigOption;
+
+/*
+ * This interface defines the api for printer configuration
+ */
+public interface ConfigurablePrinter {
+
+    /*
+     * add or update an option
+     */
+    ConfigurablePrinter addOption(ConfigOption option);
+    
+    /*
+     * Remove an option
+     */
+    ConfigurablePrinter removeOption(ConfigOption option);
+
+    /*
+     * True if an option is activated
+     */
+    boolean isActivated(ConfigOption option);
+
+    /*
+     * returns the specified option
+     */
+    Optional<ConfigOption> get(ConfigOption option);
+
+    /*
+     * returns all activated options
+     */
+    Set<ConfigOption> get();
+
+    /*
+     * returns the indentation parameters
+     */
+    Indentation getIndentation();
```

Comment:
What is special about indentation, such that it needs its own method here? (as opposed to e.g. `.get(ConfigOption.INDENTATION_STYLE)` or similar)

Reply:
Indentation is an other concept than Option so it makes a sens to distinct them.

Reply:
I think I will try to take this remark into account but maybe integrate it in another PR (asap)

Reply:
Indentation is integrated in configuration option.

Reply:
Indentation is integrated in configuration option